pkg/util/iptables/iptables.go

```diff
@@ -575,3 +576,235 @@ func IsNotFoundError(err error) bool {
 	}
 	return false
 }
+
+type RuleOption struct {
+	Arg     string
+	Negated bool
+}
+
+type Rule struct {
+	Modules []string
+	Jump    Chain
+	Comment string
+	Options map[string]RuleOption
+}
+
+type knownOption struct {
+	option      string
+	argRequired bool
+	canNegate   bool
+}
+
+func parseRule(line string, filterChains []Chain, filterChainPrefixes []string) (*Rule, Chain, error) {
+	knownOptions := []knownOption{
+		{"-A", true, false},
+		{"-j", true, false},
+		{"-m", true, false},
+		{"--comment", true, false},
+		{"--name", true, false},
+		{"--rcheck", false, true},
+		{"--seconds", true, false},
+		{"--reap", false, false},
+		{"--rsource", false, false},
+		{"--mark", true, false},
+		{"-p", true, true},
+		{"-d", true, true},
+		{"-s", true, true},
+		{"-i", true, true},
+		{"-o", true, true},
+		{"--dport", true, true},
+		{"--dst-type", true, false},
+		{"--set-xmark", true, false},
+		{"--to-destination", true, false},
+		{"--set", false, false},
+		{"--mask", true, false},
+		{"--mode", true, false},
+		{"--probability", true, false},
+		{"--physdev-is-in", false, true},
+		{"--src-type", true, true},
+	}
+
+	var rule Rule
+	var chain Chain
+
+	items := strings.Split(line, " ")
+	for i := 0; i < len(items); i++ {
+		// Check if the upcoming option is negated
+		var negated bool
+		if items[i] == "!" {
+			negated = true
+			i++
+			if i == len(items) {
+				return nil, "", fmt.Errorf("invalid iptables rule (not enough arguments): %v", line)
+			}
+		}
+
+		// Validate the option
+		var ko *knownOption
+		for _, k := range knownOptions {
+			if items[i] == k.option {
+				ko = &k
+				break
+			}
+		}
+		if ko == nil {
+			return nil, "", fmt.Errorf("Unknown iptables rule option %v", items[i])
+		}
+
+		option := RuleOption{Negated: negated}
+		if !ko.canNegate && negated {
+			return nil, "", fmt.Errorf("invalid iptables rule (can't negate option '%v'): %v", ko.option, line)
+		}
+
+		if ko.argRequired {
+			i++
+			if i == len(items) {
+				return nil, "", fmt.Errorf("invalid iptables rule (not enough arguments): %v", line)
+			}
+			if len(items[i]) == 0 {
+				return nil, "", fmt.Errorf("invalid iptables rule (empty argument): %v", line)
```

Comment:
Fixed, multiple spaces are now handled initially by a helper that splits the line, and empty quoted arguments are passed through as-is (sans quoting of course).